docs/reference/rest-api/security/bulk-create-roles.asciidoc

```diff
@@ -0,0 +1,342 @@
+[role="xpack"]
+[[security-api-bulk-put-role]]
+=== Create or update roles in Bulk API
+preview::["This functionality is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features."]
+++++
+<titleabbrev>Bulk create or update roles</titleabbrev>
+++++
+
+Bulk adds and updates roles in the native realm.
+
+[[security-api-bulk-put-role-request]]
+==== {api-request-title}
+
+`POST /_security/role/` +
+
+[[security-api-bulk-put-role-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the `manage_security` cluster
+privilege.
+
+[[security-api-bulk-put-role-desc]]
+==== {api-description-title}
+
+The role management APIs are generally the preferred way to manage roles, rather than using
+<<roles-management-file,file-based role management>>. The create
+or update roles API cannot update roles that are defined in roles files.
```

Comment:
I think this sentence is a bad copypaste from the original - should clarify for the bulk context